datalad/core/distributed/push.py

```diff
@@ -691,11 +687,34 @@ def _push_data(ds, target, content, force, jobs, res_kwargs):
                 target)
         )
         return
+
+    # it really looks like we will transfer files, get info on what annex
+    # has in store
+    ds_repo = ds.repo
+    # paths must be recoded to a dataset REPO root (in case of a symlinked
+    # location
+    annex_info_init = \
+        {ds_repo.pathobj / Path(c['path']).relative_to(ds.pathobj): c
+         for c in content} if ds.pathobj != ds_repo.pathobj else \
+        {Path(c['path']): c for c in content}
+    content = ds.repo.get_content_annexinfo(
+        # paths are taken from `content`
```

Comment:
Shouldn't that read "paths are taken from `annex_info_init`"? A bit confusing since `content` is what we assign the result to and it's not an input argument.
Note, that I take this as a comment on `paths=None`. If that is meant to be a remark on what we need the result for it should go outside the `get_content_annexinfo` call, I think.

Reply:
`annex_info_init` it built from `content`, I pushed a change to clarify that. Thx.